### PR TITLE
add .unfollowUser to user type

### PR DESCRIPTION
**Pull request by Henry-Graves**
## Describe your changes
Added backend support for unfollowing a target user.

## Issue ticket number and link
#34 

## Checklist before requesting a review
- [x] All code is linted and conforms to style guide
- [x] All necessary context is described in the PR or Issue
- [x] This branch is code and feature complete. If it is not complete, this PR is a draft.

### CODE REVIEW

File: types/user.ts [hunk 1]
```diff
@@ -76,6 +77,33 @@ export class User extends LazyObject {
...
+    return runTransaction(db, async (transaction: Transaction) => {
+      const thisSnap = await transaction.get(this.docRef!);
+      const otherSnap = await transaction.get(other.docRef!);
+
+      this.initWithDocumentData(thisSnap.data()!);
+      other.initWithDocumentData(otherSnap.data()!);
+
+      if (!containsRef(this.following!, other)) return;
+
+      this.following = this.following!.filter((user) => user !== other);
```
Comment (JakeSteinebronn) [inline types/user.ts:95]: Not quite; it's more like
`user.docRef && refEqual(user.docRef, other.docRef!)`

Comment (Henry-Graves) [reply types/user.ts:95]: @JakeSteinebronn can you please explain what you mean again?

Comment (JakeSteinebronn) [reply types/user.ts:95]: So it's like we could have two User objects which refer to the same document. In this case, the === would say they're not equal, but we really should be deleting it. And in fact, it's overwhelmingly likely for this to be the case, where you got the "other" user object from somewhere outside of your own follow list. Furthermore, it is almost impossible for you to get your own user object from the other's list of followers. However, we should remove the *corresponding* user object from the respective arrays which refer to the same document.

Comment (JakeSteinebronn) [reply types/user.ts:95]: Also, I already wrote a function for this called removeRef, it's in common

**Review by JakeSteinebronn — APPROVED** [review]
LGTM

**Pull request merged**